### PR TITLE
Add SignTool functionality to sign our analyzer assemblies and VSIXes.

### PR DESCRIPTION
**Pull request by mavasani**
Eventually we want to move away from the hard coded list of binaries to sign and our custom signtool, but this should unblock the signing for now.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/analyzer-ioperation for review

File: BuildAndTest.proj [hunk 1]
```diff
@@ -50,6 +55,10 @@
              Targets="Rebuild"/>
   </Target>
 
+  <Target Name="Sign" Condition="'$(OfficialBuild)' == 'true'">
+    <Exec Command="$(NuGetPackageRoot)\RoslynTools.Microsoft.SignTool\$(RoslynToolsMicrosoftSignToolVersion)\tools\SignTool.exe $(SignRoslynArgs) &quot;$(BinariesPath)&quot;" WorkingDirectory="$(MSBuildThisFileDirectory)" />
```
Comment (mavasani) [inline BuildAndTest.proj]: This matches the approach used by Roslyn repo: See https://github.com/dotnet/roslyn/blob/78f5f50175a051b5bf621e6f785ee1762eccb71b/src/Tools/MicroBuild/Build.proj#L36 and https://github.com/dotnet/roslyn/blob/78f5f50175a051b5bf621e6f785ee1762eccb71b/build/config/SignToolData.json and

File: BuildAndTest.proj [hunk 2]
```diff
@@ -5,13 +5,18 @@
   
   <PropertyGroup>
     <SolutionFile>$(MSBuildThisFileDirectory)RoslynAnalyzers.sln</SolutionFile>
-    <PackagingProject>$(MSBuildThisFileDirectory)\src\Packaging\Packaging.proj</PackagingProject>
+    <PackagingProject>$(MSBuildThisFileDirectory)src\Packaging\Packaging.proj</PackagingProject>
+    <BinariesPath>$(MSBuildThisFileDirectory)Binaries\$(Configuration)</BinariesPath>
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <IncludePattern Condition="'$(IncludePattern)' == ''">*.Test*.dll</IncludePattern>
-
+
```
Comment (333fred) [inline BuildAndTest.proj]: Whitespace

File: BuildAndTest.proj [hunk 3]
```diff
@@ -5,13 +5,18 @@
...
+    <BinariesPath>$(MSBuildThisFileDirectory)Binaries\$(Configuration)</BinariesPath>
     <Configuration Condition="'$(Configuration)' == ''">Debug</Configuration>
     <IncludePattern Condition="'$(IncludePattern)' == ''">*.Test*.dll</IncludePattern>
-
+    
     <!-- Emit XML in order to get structured test results -->
     <RunTestArgs>-noshadow -parallel all -xml Binaries\$(Configuration)\TestResults.xml</RunTestArgs>
 
+    <!-- Arguments to the SignTool -->
+    <SignRoslynArgs>-msbuildPath "$(MSBuildBinPath)\msbuild.exe"</SignRoslynArgs>
```
Comment (333fred) [inline BuildAndTest.proj]: I sincerely do not like the implicit config path strategy that signtool uses. Last time I had to go and make changes, finding out where the config was located, what config it was using (roslyn had 2), and why it was magically finding it wasted a bunch of my time. We can keep the config where it is, but lets please actually pass the argument to signtool with `-config <path to config>` to reduce the magic here.

**Comment by mavasani** [comment]
@333fred Addressed your feedback.

**Pull request merged**